crates/hir_ty/src/display.rs

```diff
@@ -741,7 +745,7 @@ fn write_bounds_like_dyn_trait(
                 if !first {
                     write!(f, " + ")?;
                 }
-                // We assume that the self type is $0 (i.e. the
+                // We assume that the self type is  (i.e. the
```

Comment:
o_O... I think that may have been the result of applying a code action? if there's a $0 in code, do we just let it get treated as a snippet marker?

Reply:
yup, we don't do escaping. 

:thinking: :imp: I wonder if we can bypass the whole issue by using some unicode private range  character internaly for  `$`, and do escaping in the `to_proto`?